core/connectionlistmodel.cpp

```diff
@@ -37,8 +37,10 @@ void ConnectionListModel::removeAllAddresses()
 	endRemoveRows();
 }
 
-int ConnectionListModel::indexOf(QString address)
+int ConnectionListModel::indexOf(const QString &address) const
 {
-	const QRegExp re(".*" + address + ".*", Qt::CaseInsensitive);
-	return m_addresses.indexOf(re);
+	for (int i = 0; i < address.count(); i++)
+		if (address.at(i).contains(address, Qt::CaseInsensitive))
+			return true;
```

Comment:
This should be `return i`. And of course twice `m_addresses` instead of `address`, but the compile failures already tell you that...

Reply:
sorry, pushed wrong version.

Reply:
Attention: this still says `return true` instead of `return i`!

Reply:
Apparently I'm unable to do a simple squash. Let's try again.